src/lgar.cxx

```diff
@@ -1134,7 +1137,9 @@ extern void lgar_move_wetting_fronts(double timestep_h, double *volin_cm, int wf
       double theta_new = lgar_theta_mass_balance(layer_num, soil_num, psi_cm, new_mass, prior_mass, AET_demand_cm,
 						 delta_thetas, delta_thickness, soil_type, soil_properties);
       actual_ET_demand = *AET_demand_cm;
-      
+      //done with delta_thetas and delta_thickness, cleanup memory
+      free(delta_thetas);
+      free(delta_thickness);
```

Comment:
Rather than adding `free` calls, just change the pointer variables to `std::vector<double>`, and pass down `delta_thetas.data()` for the one function argument.